src/txn/txn.c

```diff
@@ -2491,6 +2492,14 @@ __wt_txn_is_blocking(WT_SESSION_IMPL *session)
         return (0);
 #endif
 
+    /*
+     * If checkpoint is the oldest id, don't block the transaction as checkpoint may take a long
+     * time to finish. The long checkpoint might also block eviction and cause all the following
+     * transactions to rollback until it finishes.
+     */
+    if (checkpoint_id != WT_TXN_NONE && checkpoint_id == global_oldest)
+        return (0);
```

Comment:
I don't think this fix is precise enough. Generally we need checkpoints to complete, so can't kill them.
It is true that sometimes checkpoint will block eviction in particular btrees, but not in all btrees for the entire duration of the checkpoint.

I'd argue that if a test case (I think this is mostly motivated by Python test failures - is that right?) is triggering this and encountering problems due to the rollback we should reconsider how the Python test is achieving what it wants to do. i.e: overwhelming the cache to such an extent that it becomes `stuck` and looks for user transactions to kill, while a checkpoint is running for long enough to cause a problem. That Python test is probably going to be unreliable and take a long time to run. We should instead find different ways to test the functionality.

Reply:
No, the test failures are format failures. The common theme of the config that failed is that they all have slow checkpoint. They either have the debug option slow_checkpoint on or running the directio test.